v3/src/models/formula/functions/string-functions.ts

```diff
@@ -229,9 +233,51 @@ export const stringFunctions = {
   // ends with "/" it is treated as a regular expression and searched for with case sensitivity.
   wordListMatches: {
     numOfRequiredArguments: 3,
-    evaluate: (...args: FValue[]) => {
-      // todo: Implement this function
-      return "NYI"
+    evaluateRaw: (args: MathNode[], _mathjs: any, currentScope: CurrentScope) => {
+      const scope = getRootScope(currentScope)
+      const functionName = "wordListMatches"
+      const numOfReqArgs = stringFunctions.wordListMatches.numOfRequiredArguments
+      if (args.length !== numOfReqArgs) {
+        throw new Error(t("DG.Formula.FuncArgsErrorPlural.description", { vars: [ functionName, numOfReqArgs ] }))
+      }
+      [1, 2].forEach(i => {
+        if (!isConstantStringNode(args[i])) {
+          throw new Error(t("V3.formula.error.stringConstantArg", { vars: [ functionName, i + 1 ] }))
+        }
+      })
+      const [_textRefArg, dataSetTitleArg, wordAttributeNameArg, ratingAttributeNameArg] =
+        args as LookupStringConstantArg[]
+      const text = evaluateNode(args[0], scope) as string
```

Comment:
```suggestion
      const text = evaluateNode(_textRefArg, scope) as string
```

Reply:
This was giving me type problems because `_textRefArg` is cast as a `LookupStringConstantArg`.

Reply:
In that case, it's probably not worth worrying about. You could, however, type it as something like `as [any/node/whatever, LookupStringConstantArg, LookupStringConstantArg. LookupStringConstantArg]`.